File: codes/lua/1.Loops.lua
-- while loop

i1 = 1
while true do
    if (i1 == 2) then
        i1 = i1 + 1
        if (i1 > 5) then
            break
        end
        goto next
    end
    print("while loop v1, i1 is " .. i1)
    i1 = i1 + 1
    if (i1 > 5) then
        break
    end
    ::next::
end

i2 = 1
while (i2 <= 5) do
    if (i2 == 2) then
        i2 = i2 + 1
        goto next
    end
    print("while loop v2, i2 is " .. i2)
    i2 = i2 + 1
    ::next::
end

-- for loop

for i3 = 1, 5, 1 do
    if (i3 == 2) then
        goto next
    end
    print("for loop, i3 is " .. i3)
    ::next::
end

File: codes/ruby/1.Loops.rb
# while loop

i = 1
while true
    if (i == 2)
        i += 1
        break if (i > 5)
        next
    end
    print("while loop v1, i is #{i}\n")
    i += 1
    break if (i > 5)
end

i = 1
while (i <= 5)
    if (i == 2)
        i += 1
        next
    end
    print("while loop v2, i is #{i}\n")
    i += 1
end

# for loop

(1.step(5, 1)).each do |i| # we can also replace `(1.step(5, 1))` with `(1..5)`, but beware that ruby can't do `(5..1)`, use (5.step(1, -1)) instead
    next if (i == 2)
    print("for loop, i is #{i}\n")
end
